src/braket/circuits/quantum_operator.py

```diff
@@ -35,25 +40,50 @@ def __init__(self, qubit_count: int, ascii_symbols: Sequence[str]):
                 correlate a symbol with that index.
 
         Raises:
+            TypeError: `qubit_count` is not an int
             ValueError: `qubit_count` is less than 1, `ascii_symbols` are `None`, or
                 `ascii_symbols` length != `qubit_count`
         """
 
-        if qubit_count < 1:
-            raise ValueError(f"qubit_count, {qubit_count}, must be greater than zero")
-        self._qubit_count = qubit_count
+        fixed_qubit_count = self.fixed_qubit_count()
+        self._qubit_count = (
+            qubit_count if fixed_qubit_count is NotImplemented else fixed_qubit_count
+        )
```

Comment:
Would it be better to delegate this to the subclasses instead ? That is, each class passes in the `qubit_count=self.fixed_qubit_count()` instead of `qubit_count=None`. This would be much simpler and make it easier to handle other such cases in the future, if any. Trying to handle other scenarios in the QuantumOperator class could get challenging.

Reply:
Thing is, I want to enforce `GateClass.fixed_qubit_count() == GateClass(**kwargs).qubit_count`, and this was the only real way to do this in a backwards-compatible way. If each subclass does this work instead,
1. We lose that equality contract; there's no stopping the class and instants qubit counts from being different (I use "stop" loosely, as this is Python we're talking about and you can technically do anything), and
2. We basically leave it to the implementer to pass `fixed_qubit_count()` into the class' own constructor, which isn't immediately obvious

Reply:
It's a little confusing that we simply ignore the `qubit_count` value passed in, if `fixed_qubit_count` has been implemented. If we wish to enforce that `GateClass.fixed_qubit_count() == GateClass(**kwargs).qubit_count`, then could we explicitly add that check in the constructor and raise an error informing the user that fixed qubit count and qubit count values don't match, but we expect them to ?

Reply:
I can add a check to ensure that `qubit_count`, if passed in, is equal to `fixed_qubit_count()`. However, I still want the work required to implement a fixed qubit count gate to be limited to implementing `fixed_qubit_count`, and not passing that number in as well. Ideally, `qubit_count` would be the second parameter, with a default value, but that would be a breaking change for the anyone who creates new operations and uses positional arguments.

Reply:
Yes, that's true. I was thinking that if we require customers to pass in qubit_count=None, we might as well have them pass in the actual fixed qubit count. It seemed simpler, and anyone referencing the existing implementation should be able to get it right. This is not a blocker though, if you feel it won't work well.

Reply:
I think restricting `qubit_count` to equal `fixed_qubit_count` if both are nonnull is sufficient to erase the confusion. Instead of always ignoring the parameter remains optional, but still holds meaning if passed.